contracts/examples/SimpleInterestTermsContract.sol

```diff
@@ -99,20 +112,82 @@ contract SimpleInterestTermsContract is TermsContract {
         view
         returns (uint _expectedRepaymentValue)
     {
-        bytes32 parameters = debtRegistry.getTermsContractParameters(agreementId);
-        uint issuanceBlockTimestamp = debtRegistry.getIssuanceBlockTimestamp(agreementId);
+        SimpleInterestParams memory params = unpackParamsForAgreementID(agreementId);
+
+        if (timestamp <= params.startTimestamp) {
+            /* The query occurs before the contract was even initialized so the
+            expected value of repayments is 0. */
+            return 0;
+        } else if (timestamp >= params.endTimestamp) {
+            /* the query occurs beyond the contract's term, so the expected
+            value of repayment is the full principal plus interest. */
+            return params.principalPlusInterest;
+        } else {
+            uint numUnits = determineNumberOfAmortizationUnitsForTimestamp(timestamp, params);
+            return params.principalPlusInterest.mul(numUnits).div(params.termLengthInAmortizationUnits);
+        }
+    }
+
+    function determineNumberOfAmortizationUnitsForTimestamp(
```

Comment:
Nit: function title unnecessarily verbose IMO -- more readable if it was more concise (e.g. => `numAmortizationUnitsForTimestamp`)

Reply:
Agreed

Reply:
My only thought on this is that `numAmortizationUnitsForTimestamp` sounds like a property whereas `determineNumberOfAmortizationUnitsForTimestamp` sounds more like a function. 

But given that this function acts more like a property, I agree with the change.